crates/iceberg/src/spec/manifest_list.rs

```diff
@@ -931,55 +931,104 @@ mod test {
 
     use super::_serde::ManifestListV2;
 
-    #[test]
-    fn test_parse_manifest_list_v1() {
-        let path = format!(
-            "{}/testdata/simple_manifest_list_v1.avro",
-            env!("CARGO_MANIFEST_DIR")
+    #[tokio::test]
+    async fn test_parse_manifest_list_v1() {
+        let manifest_list = ManifestList {
+            entries: vec![
+                ManifestListEntry {
+                    manifest_path: "/opt/bitnami/spark/warehouse/db/table/metadata/10d28031-9739-484c-92db-cdf2975cead4-m0.avro".to_string(),
+                    manifest_length: 5806,
+                    partition_spec_id: 0,
+                    content: ManifestContentType::Data,
+                    sequence_number: 0,
+                    min_sequence_number: 0,
+                    added_snapshot_id: 1646658105718557341,
+                    added_data_files_count: Some(3),
+                    existing_data_files_count: Some(0),
+                    deleted_data_files_count: Some(0),
+                    added_rows_count: Some(3),
+                    existing_rows_count: Some(0),
+                    deleted_rows_count: Some(0),
+                    partitions: vec![],
+                    key_metadata: vec![],
+                }
+            ]
+        };
+
+        let file_io = FileIOBuilder::new_fs_io().build().unwrap();
+
+        let tmp_dir = TempDir::new().unwrap();
+        let file_name = "simple_manifest_list_v1.avro";
+        let full_path = format!("{}/{}", tmp_dir.path().to_str().unwrap(), file_name);
+
+        let mut writer = ManifestListWriter::v1(
```

Comment:
I'm bit worried about this. What will happen if we have bugs in both `ManifestListWriter` and `ManifestListReader`?

Reply:
Yes, that's a problem. But I don't see other ways to do it. I think such kinds of bugs can only be avoid during integrations tests against other systems such as spark.

Reply:
Can we call FastAvro(python) in rust to generate the file for testing🤔?

Reply:
It's possible. But I took a look at python implementation, it seems that they implemented the encoding/decoding of avro in pyiceberg, that's why they need cross validation of another library. This is not the case for rust. I think if we want to verify correctness, maybe setting up integration tests against systems like spark would be more appropriate.

Reply:
I find that avro-tool can generate an avro file by schema file and data JSON file. https://www.michael-noll.com/blog/2013/03/17/reading-and-writing-avro-files-from-the-command-line/

Reply:
Interesting, it seems that it requires java to run.

Reply:
> Interesting, it seems that it requires java to run. 

Yes. Another way is to run it in docker https://hub.docker.com/r/coderfi/avro-tools.🙈

Reply:
I did some tests and found that the json file doesn't contains all information, for example meta is missing. The docker image is not well maintained, for example last update time is 8 years ago. Given the complexity introduced, I think using integration tests with other systems maybe better?

Reply:
> I did some tests and found that the json file doesn't contains all information, for example meta is missing. The docker image is not well maintained, for example last update time is 8 years ago. Given the complexity introduced, I think using integration tests with other systems maybe better?

Thanks for your test! So seems there is no better way. Using integration tests with other systems also looks good to me. The unit test can be used to test that the reader and writer can process each other.

Reply:
Yes, keeping unit tests simple and easy to maintain. We can leave parts difficult to setup to integration tests.😁

Reply:
I have the same concern as well. Would be great to have some additional tests around Iceberg-specific things, for example, if the field-id resolution works correctly.